app/controllers/concerns/starter_code.rb

```diff
@@ -7,8 +7,8 @@ def starter_code_repository_id(repo_name)
 
     if repo_name =~ %r{^[a-zA-Z0-9_-]+\/[a-zA-Z0-9_-]+$}
       begin
-        github_repository = GitHubRepository.new(current_user.github_client, nil)
-        github_repository.repository(repo_name).id
+        github_repository = GitHubRepository.new(id: nil, access_token: current_user.access_token)
+        github_repository.repository(full_name: repo_name).id
```

Comment:
I don't understand why we're using the `GitHubRepository#repository` instance method here. I think this would be more logical as a call to an instance method on an API client that was initialized with the correct token.

The way it reads now is "create a nil GitHub Repository and ask that repository for the id of some other repository" which is awful roundabout.